core/errors.ts

```diff
@@ -48,14 +48,22 @@ export class Exception extends Error {
 }
 
 /** Pretty-print an Error or Exception instance */
-export function printError(error: Error, caused = false) {
+export function printError(error: unknown, caused = false) {
```

Comment:
I didn't know how to test this without refactoring out a function that returns a string instead of logging to the console.

Reply:
Thanks for fixing this. You're right, maybe this function should return a string.
Do you want to change it? I think replacing all `console.log(whatever)` with `logs.push(whatever)` and then return `logs.join("\n")` should be enough.
And if you cannot do it, don't worry, I will do it after mergin this.

Reply:
I looked into it a bit and it's a bit more involved because some are console.logs and some are console.error (I think probably all should just be switch to console.error)? There also needs some care taken for converting the statements that use multiple arguments. Maybe it's not worth it to change at the moment? (though it probably would be good for unit testing)

Reply:
All right, no worries. I'm merging this and will improve it later. Thanks!